homeassistant/components/shelly/cover.py

```diff
@@ -110,3 +139,61 @@ def _update_callback(self) -> None:
         """When device updates, clear control result that overrides state."""
         self.control_result = None
         super()._update_callback()
+
+
+class RpcShellyCover(ShellyRpcEntity, CoverEntity):
+    """Entity that controls a cover on RPC based Shelly devices."""
+
+    _attr_device_class = CoverDeviceClass.SHUTTER
+
+    def __init__(self, wrapper: RpcDeviceWrapper, id_: int) -> None:
+        """Initialize rpc cover."""
+        super().__init__(wrapper, f"cover:{id_}")
+        self._id = id_
+        self._attr_supported_features: int = SUPPORT_OPEN | SUPPORT_CLOSE | SUPPORT_STOP
+        if self.wrapper.device.status[self.key]["pos_control"]:
+            self._attr_supported_features |= SUPPORT_SET_POSITION
+
+    @property
+    def is_closed(self) -> bool | None:
+        """If cover is closed."""
+        if not self.wrapper.device.status[self.key]["pos_control"]:
```

Comment:
You could shorten this a bit by making a 
```python3
    @property
    def device_status(self) -> dict:
        return self.wrapper.device.status[self.key]
```
and then...
```suggestion
        if not self.device_status["pos_control"]:
```

Reply:
I am going to do this in the base entity (`ShellyRpcEntity`) since other platform will reuse it. Thanks